ScrapeRepos.ts

```diff
@@ -0,0 +1,356 @@
+      developers.push({ name, username });
+    });
+
+    // get the repo name
+    $('h1.h4.lh-condensed a').each((i: number, el: cheerio.Element) => {
+      const repo: string | undefined = $(el).attr('href')?.substr(1);
+      // check if the repo exists
+      if (repo) {
+        const split = repo.split("/");
+        const username = split[0];
+        const repoName = split[1];
+        developerRepos.push({ username, repo: repoName });
+      }
+    });
+
+    // correctly merge the two arrays
+    const repo = "";
+    const developersWithRepos = developers.map(developer => {
+      const matchingRepo = developerRepos.find(repo => repo.username === developer.username);
+      return { ...developer, ...(matchingRepo || {repo}) };
+    });
+
+    return developersWithRepos
+  }).catch((error: any) => {
+    console.log(error);
+  });
+}
+
+
+/** Main function to test the functionality of the different methods
+ * and how to correctly call them and what the intended workflow is about
+ * @param Github and OpenAI token still have to be added manually
+ */ 
+async function main() {
+  // Choose whether to scrape daily, weekly or monthly information
+  const timeMode = 0
+  console.log("TimeMode: " + timeMode)
+
+  const trendingSplit: string[] | undefined= await fetchRepos(timeMode);
+  
+  // your personal GitHub authToken
+  const authToken = `Bearer XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX`;
+  // //Authorization: "Bearer " + process.env.OPENAI_API_KEY,
+  const OPENAI_API_KEY = "sk-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
+  
+  // check if any repos were actually found
+  if(!trendingSplit){
+      console.log("Repos couldn't be scraped")
+      process.exit()
+  }
+
+  for(let i = 0; i < trendingSplit.length / 2; i++){
+    const owner = trendingSplit[2*i];
+    const name = trendingSplit[2*i+1];
+    const query = `query {
+      repository(owner: "${owner}", name: "${name}") {
+        name    
+        description
+        url
+        createdAt
+        stargazers {
+          totalCount
+        }
+        forks {
+          totalCount
+        }
+        primaryLanguage {
+          name
+        }
+      }
+    }`;
+
+    const output = await getRepoInfo(query, authToken)
+    console.log(output)
+
+    // check if the repo has more than a 1k stars
+    const stars = output.stargazers.totalCount
+    if(stars < 1000){
+      console.log("Wouldn't provide this entity to the frontend")
+    }
+
+    const readme = await getReadme(owner, name)
+
+    if(readme != null){
+      // call openai api
+      const answer = await openAIRequestTurbo(readme, OPENAI_API_KEY)
+      console.log(answer)
+    }
+
+    // get the star history of the repo
+    // await fetchRepositoryStars(owner, name, authToken)
+    const result = await fetchRepositoryStars(owner, name, authToken)
+    console.log(result)
+  }
+  // get the developers
+  const developers = await fetchDevelopers(timeMode)
+  console.log(developers)
+  
+}
+main()
```

Comment:
Why did you use main? If you put this here for testing, please remove before opening the pr.